app/services/force/units_service.rb

```diff
@@ -0,0 +1,46 @@
+module Force
+  # encapsulate all Salesforce Short Form Application querying functions
+  class UnitsService < Force::Base
+    DRAFT = 'Draft'.freeze
+    FIELD_NAME = :units
+    FIELDS = load_fields(FIELD_NAME).freeze
+
+    def available_units_for_application(listing_id, application_id)
+      sub_query = builder.from(:Lease__c)
+                         .select(:Unit__c)
+                         .where_eq('Application__r.Listing__c', listing_id, :string)
+                         .where("Application__c != '#{application_id}'")
+                         .to_soql
+
+      builder.from(:Unit__c)
+             .select('Id, Unit_Number__c ')
+             .where_eq('Listing__c', listing_id, :string)
+             .where("Id NOT IN (#{sub_query})")
+             .transform_results { |results| massage(results) }
+             .query
+             .records
+    end
+
+    def unassinged_units_by_listing(listing_id)
```

Comment:
typo unassigned

Reply:
does a unit know if it's assigned or not assigned? or is it just checking to see if an application has a unit assigned to it?

Reply:
are these in use anywhere?

Reply:
This is not being used. I created and then I had to use a different query. I left it just in case for future use in case we need it since the query is not so simple.

Reply:
can you add some documentation to the functions to make it clear what they're doing?

Reply:
👍 docs added